bootstrap/bootstrap.go

```diff
@@ -814,6 +818,18 @@ func (b *Bootstrap) CheckoutPhase() error {
 
 	b.shell.Headerf("Preparing working directory")
 
+	// If we have a blank repository then use a temp dir for builds
+	if b.Config.Repository == "" {
+		buildDir, err := ioutil.TempDir("", "bootstrap-builds")
```

Comment:
If this is going to be semi-visible, maybe "buildkite" over "bootstrap"? (We could possibly even include the build id or something to allow better backtracking if one notices dirs hanging around, but I don't know whether that's worth the bother.)